crm-data-components/src/app/extensions/ActionsExtension.tsx

```diff
@@ -0,0 +1,169 @@
+import React, { useState } from "react";
+import {
+  Divider,
+  Text,
+  Flex,
+  hubspot,
+  Form,
+  Select,
+} from "@hubspot/ui-extensions";
+import {
+  CrmPropertyList,
+  CrmCardActions,
+  CrmActionButton,
+} from "@hubspot/ui-extensions/crm";
+
+// Define the extension to be run within the Hubspot CRM
+hubspot.extend(({ context, runServerlessFunction, actions }) => (
+  <Extension
+    context={context}
+    runServerless={runServerlessFunction}
+    sendAlert={actions.addAlert}
+  />
+));
+
+const scenariosToPropertiesMap: { [key: string]: string[] } = {
+  contact: [
+    "jobtitle",
+    "industry",
+    "date_of_birth",
+    "hs_language",
+    "favorite_color",
+  ],
+  engagement: [
+    "hs_sa_first_engagement_date",
+    "hs_latest_source_data_1",
+    "engagements_last_meeting_booked",
+    "hs_email_sends_since_last_engagement",
+    "hs_time_to_first_engagement",
+  ],
+  deals: [
+    "hs_buying_role",
+    "num_associated_deals",
+    "recent_deal_amount",
+    "recent_deal_close_date",
+  ],
+  nps: [
+    "hs_feedback_show_nps_web_survey",
+    "hs_feedback_last_survey_date",
+    "hs_feedback_last_nps_rating",
+    "hs_feedback_last_nps_follow_up",
+  ],
+};
+
+const options: Array<{ label: string; value: string }> = [
+  { label: "Contact", value: "contact" },
+  { label: "Engagement", value: "engagement" },
+  { label: "Deal Summary", value: "deals" },
+  { label: "NPS", value: "nps" },
+];
+
+const Extension = ({ context, runServerless, sendAlert }) => {
+  const [scenario, setScenario] = useState<any>("engagement");
```

Comment:
Should this be a string if using ts?
```suggestion
  const [scenario, setScenario] = useState<string>("engagement");
```